common/lib/dependabot/errors.rb

```diff
@@ -92,6 +92,11 @@ def self.fetcher_error_details(error)
           "rate-limit-reset": T.cast(error, Octokit::Error).response_headers["X-RateLimit-Reset"]
         }
       }
+    when BadRequirementError
+      {
+        "error-type": "illformed_requirement",
+        "error-detail": { message: error.message }
+      }
```

Comment:
This block of code already existed in `updater_error_details` but this error occurs during the `fetch` phase of a NuGet update so it was simply copied here, too.  This is a well-known error with a known shape.